app/controllers/api/functions.php

```diff
@@ -1523,7 +1523,8 @@
             throw new Exception(Exception::FUNCTION_NOT_FOUND);
         }
 
-        $runtimes = Config::getParam('runtimes', []);
+        $version = $function->getAttribute('version', 'v2');
+        $runtimes = Config::getParam($version === 'v2' ? 'runtimes-v2' : 'runtimes', []);
```

Comment:
Dont we also need to update it here
https://github.com/appwrite/appwrite/blob/fc3ff188b143d55bc7a6c1fda3072b3913f89a6b/app/controllers/api/vcs.php#L478

and here
https://github.com/appwrite/appwrite/blob/fc3ff188b143d55bc7a6c1fda3072b3913f89a6b/app/controllers/api/vcs.php#L558

Reply:
That's for runtime auto-detection IIRC. If you are connecting v2 function to git, you are weird.
Also, if detection fails, Console gives you dropdown so you can pick

Reply:
Similiarly, listRuntimes endpoint will return v3 stuff even if your function is v2. But same rule, you shouldn't really be changing your runtime if your function is v2.

If you are actively working on your function (connecting it to git and upgrading runtime), you should also upgrade to v3. V2 support should be purely to not break existing things

Reply:
lets say someone has an existing v2 function and only want to connect it to git without really ugrading the code/syntax.

In that case we need to account for v2 runtimes here too right ?

Reply:
library appwrite/runtimes isn't really aware of v2 list anyway.
Let's say there was V2 Node 16, and now there is V3 Node 18..
In Appwrite 1.4, it will only know "Node 18". So it will think "v2 Node 18" is a thing, even if it isn't.

If we want to fully be aware of v2 runtimes, we need to introduce and maintain them here: https://github.com/appwrite/runtimes

I personally don't really see value in that. If you want to connect git, you actively work on project. You should migrate to V3.

Also, from technical point of view, this VCS endpoint isn't aware of function. It just gets repo, and does detection. We would need to add `version` parameter to the VCS endpoint. I really dont like touching code this much for simple backwards compatibility.